kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericMatrix.js

```diff
@@ -0,0 +1,308 @@
+/**
+ * Output widget to vizualize ExpressionMatrix object.
+ * Pavel Novichkov <psnovichkov@lbl.gov>
+ * @public
+ */
+
+define ([
+    'uuid',
+    'jquery',
+    'kbwidget',
+    'kbaseAuthenticatedWidget',
+    'kbaseTabs',
+    'narrativeConfig',
+		'kbase-generic-client-api',
+    'bluebird',
+    // For effect
+    'bootstrap',
+    'jquery-dataTables',
+    'datatables.net-buttons',
+    'datatables.net-buttons-bs',
+    'datatables.net-buttons-html5',
+    'datatables.net-buttons-print'
+], function(
+    Uuid,
+    $,
+    KBWidget,
+    kbaseAuthenticatedWidget,
+    kbaseTabs,
+    Config,
+    GenericClient,
+    Promise
+) {
+    'use strict';
+
+    return KBWidget({
+        name: 'kbaseGenericMatrix',
+        parent : kbaseAuthenticatedWidget,
+        version: '1.0.2',
+        options: {
+            matrixID: null,
+            workspaceID: null,
+            loadingImage: Config.get('loading_gif')
+        },
+
+        // Prefix for all div ids
+        pref: null,
```

Comment:
The `pref` variable probably isn't necessary. It doesn't look like you're using it to search for ids or anything to modify or update. And as long as you limit your search to children of `this.$elem`, it's not needed at all. Doesn't necessarily need to be changed, though.